### PR TITLE
Fixes crash on JSON syntax error in FMNetworkManager.processAndLogError, closes MW-551

### PR DESCRIPTION
**Pull request by bastinagual**

### CODE REVIEW

**Comment by jpvsilva-fss** [comment]
Besides that try/catch, I found it crashing right after the processAndLogError call with the same exception. At lines 68-70 of FMNetworkManager we are repeating the same object creation from Json. We would need to improve this.

**Comment by bastinagual** [comment]
Right! Seems there are three more places to change this then.

**Comment by bastinagual** [comment]
Made some changes, should not crash on errors anymore

**Review by jpvsilva-fss — APPROVED** [review]

File: FantasmoSDK/app/src/main/java/com/fantasmo/sdk/network/FMNetworkManager.kt [hunk 1]
```diff
@@ -237,27 +233,34 @@ class FMNetworkManager(
...
                 when (networkResponse.statusCode) {
                     404 -> {
                         errorMessage = "Resource not found"
                     }
+                    403 -> {
+                        errorMessage = "Forbidden"
+                    }
                     401 -> {
-                        errorMessage = "${response.message} Authentication error"
+                        errorMessage = "$debugMessage Authentication error"
```
Comment (GrayedFox) [inline FantasmoSDK/app/src/main/java/com/fantasmo/sdk/network/FMNetworkManager.kt:247]: don't these need `${curlyBrackets}` to interpolate properly?

Comment (bastinagual) [reply FantasmoSDK/app/src/main/java/com/fantasmo/sdk/network/FMNetworkManager.kt:247]: The brackets are only needed if the thing one wants to print is more than just a variable name

**Review by GrayedFox — COMMENTED** [review]
lgtm

**Pull request merged**